src/service/chat.js

```diff
@@ -206,6 +206,71 @@ const moveChatToAnon = async ({
   }
 };
 
+const initChatFromPost = async ({source, id}) => {
+  let payload = {source};
+  switch (source) {
+    case 'post':
+      payload = {...payload, postId: id};
+      break;
+    case 'comment':
+      payload = {...payload, commentId: id};
+      break;
+    default:
+      break;
+  }
+
+  try {
+    const response = await api.post('/chat/init-chat-from-post', payload);
+    if (response?.status === 200) {
+      return Promise.resolve(response.data);
+    }
+
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `initChatFromPost` function could be simplified by removing the switch statement and using a computed property name instead. This would reduce the complexity of the code and make it more maintainable.
```diff
-  let payload = {source};
-  switch (source) {
-    case 'post':
-      payload = {...payload, postId: id};
-      break;
-    case 'comment':
-      payload = {...payload, commentId: id};
-      break;
-    default:
-      break;
-  }
+  const key = source === 'post' ? 'postId' : 'commentId';
+  const payload = {source, [key]: id};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->